File: blind-75/arrays-and-hashing/two-sum/go/two_sum.go
package twosum

func TwoSum(nums []int, target int) []int {
	diff := make(map[int]int)
	var ans []int

	for i := 0; i < len(nums); i++ {
		if _, exist := diff[target-nums[i]]; exist {
			ans = append(ans, diff[target-nums[i]], i)

			break
		}

		diff[nums[i]] = i
	}

	return ans
}

File: blind-75/arrays-and-hashing/two-sum/python/two_sum.py
def two_sum(nums: list[int], target: int) -> list[int]:
  diff = {}
  ans = []

  for i in range(len(nums)):
    if target - nums[i] in diff:
      ans.append(diff.get(target - nums[i]))
      ans.append(i)

      break

    diff[nums[i]] = i

  return ans